lib/sshutils/sftp/sftp.go

```diff
@@ -561,10 +564,19 @@ func (c *Config) transfer(ctx context.Context) error {
 }
 
 // transferDir transfers a directory
-func (c *Config) transferDir(ctx context.Context, dstPath, srcPath string, srcFileInfo os.FileInfo) error {
+func (c *Config) transferDir(ctx context.Context, dstPath, srcPath string, srcFileInfo os.FileInfo, visited map[string]struct{}) error {
+	realSrcPath, err := c.srcFS.RealPath(srcPath)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if _, ok := visited[realSrcPath]; ok {
```

Comment:
```suggestion
	if visited == nil {
		visited = make(map[string]struct{})
	}
	if _, ok := visited[realSrcPath]; ok {
```
Any reason not to automatically initialize `visited` if we're passed a `nil`? That way callers don't have to allocate an empty map and we shouldn't ever end up with a surprise panic on line 576